lib/s3routes/routesUtils.js

```diff
@@ -402,6 +411,21 @@ const routesUtils = {
             return XMLResponseBackend.errorResponse(errCode, response, log,
                                                     resHeaders);
         }
+        if (dataLocations !== null && !response.headersSent) {
+            // sanity check of content length against individual data
+            // locations to fetch
+            const contentLength = resHeaders && resHeaders['Content-Length'];
+            if (contentLength !== undefined &&
+                !_contentLengthMatchesLocations(contentLength,
+                                                dataLocations)) {
+                log.error('logic error: total length of fetched data ' +
```

Comment:
May be remove `logic error:` and keep the rest of the message?

Reply:
ok

Reply:
Well, maybe not going to change it since master branch is already merged.